submission/submitter/src/main/kotlin/ac/uk/ebi/biostd/submission/submitter/SubmissionSubmitter.kt

```diff
@@ -17,10 +22,39 @@ class SubmissionSubmitter(
         submission: ExtendedSubmission,
         files: FilesSource,
         context: PersistenceContext
+
     ): Submission {
-        validators.forEach { validator -> validator.validate(submission, context) }
-        processors.forEach { processor -> processor.process(submission, context) }
-        filesHandler.processFiles(submission, files)
+        val validationNodes = mutableListOf<ValidationNode>()
+
+        validators.map {
```

Comment:
I think we can do this code more declarative(or functional) and keep logic clean (right now is obfuscated by error handling) which is not the purpose of this class, what we can do here is. Collect all exceptions in a `Try` and accumulate them, then we check if not empty in order to trigger the error or not.  Check https://arrow-kt.io/docs/arrow/core/try/